src/libraries/System.Security.Cryptography/tests/X509Certificates/CertificateCreation/PrivateKeyAssociationTests.cs

```diff
@@ -844,6 +846,82 @@ public static void CheckCopyWithPrivateKey_MLKem_OtherMLKem_DecapsulationKey()
             }
         }
 
+        [ConditionalFact(typeof(SlhDsa), nameof(SlhDsa.IsSupported))]
+        public static void CheckCopyWithPrivateKey_SlhDsa()
+        {
+            string certPem = PemEncoding.WriteString("CERTIFICATE", SlhDsaTestData.IetfSlhDsaSha2_128sCertificate);
+
+            using (X509Certificate2 pubOnly = X509Certificate2.CreateFromPem(certPem))
+            using (SlhDsa privKey = SlhDsa.ImportPkcs8PrivateKey(SlhDsaTestData.IetfSlhDsaSha2_128sPrivateKeyPkcs8))
+            using (X509Certificate2 wrongAlg = X509CertificateLoader.LoadCertificate(TestData.CertWithEnhancedKeyUsage))
+            {
+                CheckCopyWithPrivateKey(
+                    pubOnly,
+                    wrongAlg,
+                    privKey,
+                    SlhDsaTestData.AlgorithmsRaw.Select(algo => (Func<SlhDsa>)(() => SlhDsa.GenerateKey(algo))),
```

Comment:
I don't think we need everything here. EC-DSA gets by with the most common 3.  A top, a bottom, and one from the middle seem sufficient to me.  "Worst case", both tops, both bottoms, and two different sizes from the middle (one SHA, one SHAKE)

Reply:
My candidate set would be
* SLH-DSA-SHA2-128s
* SLH-DSA-SHAKE-192s
* SLH-DSA-SHAKE-256f 

Each step changes two things.  (SHA2/128) => (SHAKE, 192), then (192, s) => (256, f).

Reply:
`SLH-DSA-SHA2-128s` and `SLH-DSA-SHAKE-256f` for sure because they're the first and last in the OID range, but probably `SLH-DSA-SHA2-192f` instead of `SLH-DSA-SHAKE-192s` since it's ~30x faster.

Reply:
This particular test is slow(-ish) but not because we're using all 12 algos. The 12 algos don't sign, only `IetfSlhDsaSha2_128sPrivateKey` does. However, `IetfSlhDsaSha2_128sPrivateKey` is an 's' algo and the test signs 5 times so it ends up being slow anyway.

These tests increase the runtime of the System.Security.Cryptography test suite by about 30% which is a lot, but I think it's worth having them until we finish more features and the Windows implementation. Once those are done then I'll make a pass over all the tests to limit them to smaller representative data sets.

Reply:
This specific line of code is just about making sure CopyWithPrivateKey rejects bad keys.  So it needs one key of the same parameter set as the pubOnly cert, one key of a different one.  An extra one for feel goodsies is also fine.

The PFX load, et cetera, are presumably where the real time is.  But this line should just be keygen (should be fast for all parameter sets) and export public key (to compare with the one from pubOnly), which I'd hope is fast.

My original "trim this down" probably applies elsewhere, but I agree that we want to sanity-cover each OID, therefore each parameter set.